framework/Models/PostProcessors/PostProcessorInterface.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on April 6, 2021
+
+@author: wangc
+"""
+
+#External Modules------------------------------------------------------------------------------------
+import copy
+import abc
+import os
+#External Modules End--------------------------------------------------------------------------------
+
+#Internal Modules------------------------------------------------------------------------------------
+from BaseClasses import BaseInterface
+from utils import InputTypes, InputData
+import Runners
+#Internal Modules End--------------------------------------------------------------------------------
+
+class PostProcessorInterface(BaseInterface):
+  """
+    Base class for other postprocessor interfaces (i.e., BasicStatistics, ETImporter).
+  """
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    spec = super().getInputSpecification()
+    spec.addParam("subType", InputTypes.StringType, True)
+    return spec
+
+  def __init__(self):
+    """
+      Constructor
+      @ In, runInfoDict, dict, the dictionary containing the runInfo (read in the XML input file)
+      @ Out, None
+    """
+    super().__init__()
+    self.inputCheckInfo  = []     # List of tuple, i.e input objects info [('name','type')]
+    self.action = None            # action
+    self.workingDir = ''          # path for working directory
+    self.printTag = 'PostProcessorInterface'
+    self.outputDataset  = False # True if the user wants to dump the outputs to dataset
+    self.validDataType = ['PointSet','HistorySet'] # The list of accepted types of DataObject
```

Comment:
should not be this added in the Models.py validation? in addition,if we have those here why in the other PR we still check for "Databases"?

Reply:
This is for specific post-processors. They can accept only PointSet or both PointSet and HistorySet. I do not see the option that we can check at the Model level. It is PP-dependent options.

Reply:
see comment below...the "validateDict" approach can be expanded to the entity baseInterface classes. Right?